cache/store.go

```diff
@@ -0,0 +1,98 @@
+// Copyright 2025 The etcd Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package cache
+
+import (
+	"bytes"
+	"fmt"
+	"sort"
+	"sync"
+
+	"go.etcd.io/etcd/api/v3/mvccpb"
+	clientv3 "go.etcd.io/etcd/client/v3"
+)
+
+type store struct {
+	mu        sync.RWMutex
+	kvs       map[string]*mvccpb.KeyValue
+	latestRev int64
+}
+
+func (s *store) Get(startKey, endKey []byte) ([]*mvccpb.KeyValue, int64) {
+	s.mu.RLock()
+	defer s.mu.RUnlock()
+
+	if len(endKey) == 0 { // fast path: single key lookup
+		if kv, ok := s.kvs[string(startKey)]; ok {
+			return []*mvccpb.KeyValue{kv}, s.latestRev
+		}
+		return nil, s.latestRev
+	}
+
+	var out []*mvccpb.KeyValue
+	for _, kv := range s.kvs { // range / prefix scan
+		k := kv.Key
+		switch {
+		case len(endKey) == 1 && endKey[0] == 0: // prefix & fromKey
```

Comment:
Can we modify the signature and pass in `op` and just check `op.IsWithPrefix()`?